### PR TITLE
Beacons are no longer radios

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do

Refactors beacons from /obj/item/radio/beacon to /obj/item/beacon
Renames from "Tracking Beacon" to "tracking beacon"

## Why It's Good For The Game
Goodbye shitty type pathing

## Testing

Could use syndi beacons to call in bombs, etc
Could teleport with hand tele and normal teleporter
Jaunters seem to still work too

## Changelog
:cl:
fix: Tracking breacons no longer have a hint in their description saying you can turn on their "broadcasting"
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/radio/beacon.dm [hunk 1]
```diff
@@ -1,63 +1,69 @@
...
+	w_class = WEIGHT_CLASS_SMALL
+
+	materials = list(MAT_METAL = 200, MAT_GLASS = 100)
 	var/syndicate = FALSE
 	var/area_bypass = FALSE
 	var/cc_beacon = FALSE //set if allowed to teleport to even if on zlevel2
 
-/obj/item/radio/beacon/New()
+/obj/item/beacon/New()
 	..()
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/beacon.dm:20]:
```suggestion
/obj/item/beacon/Initialize()
	. = ..()
```
Could you check if this still works?

File: code/game/objects/items/devices/radio/beacon.dm [hunk 2]
```diff
@@ -1,63 +1,69 @@
...
-	GLOB.beacons += src
+	GLOB.beacons |= src
 
-/obj/item/radio/beacon/Destroy()
+/obj/item/beacon/Destroy()
 	GLOB.beacons -= src
 	return ..()
 
-/obj/item/radio/beacon/emag_act(user as mob)
+/obj/item/beacon/emag_act(user as mob)
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/beacon.dm:27]:
```suggestion
/obj/item/beacon/emag_act(user)
```

File: code/game/objects/items/devices/radio/beacon.dm [hunk 3]
```diff
@@ -1,63 +1,69 @@
...
-/obj/item/radio/beacon/hear_talk()
-	return
 
 /// Probably a better way of doing this, I'm lazy.
-/obj/item/radio/beacon/bacon
+/obj/item/beacon/bacon
 
-/obj/item/radio/beacon/bacon/proc/digest_delay()
+/obj/item/beacon/bacon/proc/digest_delay()
 	QDEL_IN(src, 600)
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/beacon.dm:39]:
```suggestion
	QDEL_IN(src, 60 SECONDS)
```

File: code/game/objects/items/devices/radio/beacon.dm [hunk 4]
```diff
@@ -1,63 +1,69 @@
...
 	emagged = TRUE
 
 // SINGULO BEACON SPAWNER
-/obj/item/radio/beacon/syndicate
+/obj/item/beacon/syndicate
 	name = "suspicious beacon"
 	desc = "A label on it reads: <i>Activate to have a singularity beacon teleported to your location</i>."
 	origin_tech = "bluespace=6;syndicate=5"
 	syndicate = TRUE
 	var/obj/machinery/computer/syndicate_depot/teleporter/mycomputer
```
Comment (DGamerL) [inline code/game/objects/items/devices/radio/beacon.dm:51]: This variable seems to be unused apart from the `Destroy()`, and my compiler doesn't catch any undefined vars if I remove this

**Review by DGamerL — COMMENTED** [review]
Looks good

File: code/game/objects/items/devices/radio/beacon.dm (hunk 4 above)
Comment (Contrabang) [reply code/game/objects/items/devices/radio/beacon.dm:51]: Out of Scope

File: code/modules/projectiles/projectile/special_projectiles.dm [hunk 5]
```diff
@@ -290,7 +290,7 @@
 	damage = 0
 	nodamage = 1
 	alwayslog = TRUE
-	var/obj/item/radio/beacon/teleport_target = null
+	var/obj/item/beacon/teleport_target = null
```
Comment (DGamerL) [inline code/modules/projectiles/projectile/special_projectiles.dm:293]:
```suggestion
	var/obj/item/beacon/teleport_target
```

**Review by DGamerL — APPROVED** [review]

**Review by lewcc — APPROVED** [review]

**Pull request merged**